openedx/core/djangoapps/user_api/accounts/image_helpers.py

```diff
@@ -55,7 +55,12 @@ def _make_url(size):
         url = storage.url(
             _get_profile_image_filename(name, size, file_extension=file_extension)
         )
-        return '{}?v={}'.format(url, version) if version is not None else url
+        # Return the URL, with the "v" parameter added as its query
+        # string with "?v=". If the original URL already includes a
+        # query string (such as signed S3 URLs), append to the query
+        # string with "&v=" instead.
+        separator = '&' if '?' in url else '?'
+        return '{}{}v={}'.format(url, separator, version) if version is not None else url
```

Comment:
FWIW, this line is odd enough that I would have changed it to:
```
if version is not None:
    separator = '&' if '?' in url else '?'
    url += "{}v={}".format(separator, version)
return url
```

Reply:
Fair point. 🙂